### DISCUSSION TITLE
importing modules from node_modules in a vue component

### DISCUSSION THREAD

**Discussion by OriLapushner** [nuxt/nuxt#29689] [Questions]
I am new to nuxt, and I'm trying to create a new project with typescript.
package.json versions:
    "nuxt": "^3.13.2"
    "typescript": "^5.6.3"
    "vue-tsc": "^2.1.6"

I am using vite and VS code.
When I try to import something from node_modules in a ,vue file where the script is setup to lang="ts"  I get this error:
Cannot find module 'my-import-path'. Did you mean to set the 'moduleResolution' option to 'node', or to add aliases to the 'paths' option?
when I remove the lang="ts" the error no longer exists.
when I import from a .ts file there is no error.
It's possible to import to a .ts file and then import it to the .vue file but this just seems like an ugly work around.
the project works regardless of the error but I do get it in my problems tab in VS code.

in order to reproduce my issue follow this steps:
run in the command line:
npx nuxi@latest init my-app
cd my-app
npm install --save-dev vue-tsc typescript

now install any package in my case @heroicons/vue:
npm i @heroicons/vue

try to import the module inside app.vue:
<template>
	<div>
		<p>test</p>
	</div>
</template>

<script setup lang="ts">
import { BeakerIcon } from "@heroicons/vue/24/solid";
</script>

the error you get should be:
Cannot find module '@heroicons/vue/24/solid'. Did you mean to set the 'moduleResolution' option to 'node', or to add aliases to the 'paths' option?

I've tried various solutions I found online and none worked.
My guess is that I need to configure aliases within vue files which I cannot seem to successfully achieve.

**Comment by Lumethys** [comment]
Read the docs. Nuxt use autoimport so dont explicitly import your dependency like that

**Reply by flyxiong819** [reply]
I got the same problem. Not with nuxtjs, but with vue2.7 + typescript.
Have you solved it?

**Reply by flyxiong819** [reply]
I fix my problem. I just install `@vue/cli-plugin-typescript` and change entry file ‘main.js’ to 'main.ts'. It's OK for me.
